### PR TITLE
Fixes #22607 - add rex ssh to backup

### PR DESCRIPTION
**Pull request by chris1984**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.17
* [ ] 1.16
* [ ] 1.15

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by chris1984** [comment]
@akofink, unfortunately, yes, since we do not encrypt any part of the backup. I can raise a docs bug for downstream and create a doc pr upstream to make sure the backup is safeguarded if that would suffice?

**Comment by akofink** [comment]
We should definitely have a conversation about it. I like the idea of adding documentation that states that this could compromise security, and I also don't see why we can't incorporate an option for encryption of the sensitive data in the future. We obviously don't need to encrypt pulp or anything.

**Comment by evgeni** [comment]
Well, a rouge backup can already compromise any machine by injecting puppet modules or rpms. Or telling a random rex proxy to execute stuff via ssh. So this isn't making it much worse (ssh is easier to start than a puppet master, though ;))

**Comment by akofink** [comment]
True, not saying we should block this PR. :)

File: packages/katello/katello/backup.rb [hunk 1]
```diff
@@ -87,7 +87,8 @@ def configure_configs
         '/opt/puppetlabs/puppet/cache/foreman_cache_data',
         '/opt/puppetlabs/puppet/ssl/',
         '/var/lib/puppet/foreman_cache_data',
-        '/var/lib/puppet/ssl'
+        '/var/lib/puppet/ssl',
+        '/usr/share/foreman-proxy/.ssh/'
```
Comment (ekohl) [inline packages/katello/katello/backup.rb]: This does make me wonder: why is it writing in ```/usr```? FHS says this shouldn't happen and as an admin I'd be surprised about this behavior. Why doesn't it live in ```/var/lib/foreman-proxy```?

Comment (stbenjam) [reply packages/katello/katello/backup.rb]: foreman-proxy $HOME is /usr/share/foreman-proxy

Comment (ekohl) [reply packages/katello/katello/backup.rb]: It sounds like something would should fix, possibly via a symlink.

**Comment by chris1984** [comment]
@akofink 

Downstream:

https://bugzilla.redhat.com/show_bug.cgi?id=1547650

Upstream:

https://projects.theforeman.org/issues/22648

**Comment by chris1984** [comment]
@akofink can this get a merge?

**Comment by akofink** [comment]
I have no perms here, but yes! @jlsherrill?

**Comment by evgeni** [comment]
wonder if I should make it a habit pretending to be @jlsherrill? ;-)

**Pull request merged**